clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -715,45 +717,32 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(int output_
         cur_range = input.regions_of_interest[input.cur_region];
     }
 
-    if (cur_instr >= cur_range.start_instruction)
+    // If we're within one and already skipped, just exit to avoid re-requesting a skip
+    // and making no progress (we're on the inserted timetamp + cpuid and our cur instr
+    // count isn't yet the target).
+    if (input.in_cur_region && cur_instr >= cur_range.start_instruction - 1)
         return sched_type_t::STATUS_OK;
 
     VPRINT(this, 2, "skipping from %" PRId64 " to %" PRId64 " instrs for ROI\n",
            cur_instr, cur_range.start_instruction);
+    input.in_cur_region = true;
```

Comment:
nit: If it doesn't break any logic: it'd be better to set this bool only after we actually skip the instrs at L734.

Reply:
Done.